### PR TITLE
AB#87178 Added notifications for InviteUser and RemoveUser actions

### PR DESCRIPTION
**Pull request by Antek012**

### CODE REVIEW

File: src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs [hunk 1]
```diff
@@ -43,6 +47,8 @@ public async Task<OperationResult> Handle(InviteUserToOrganisationCommand reques
             return ex.OperationResult;
         }
 
+        await _notificationService.Publish(new UserInvitedNotification($"{request.FirstName} {request.LastName}"));
```
Comment (jjankowski87) [inline src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs:50]: decouple notification logic from command handlers by domain events

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs (hunk 1 above)
Comment (Antek012) [reply src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs]: done

File: src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs [hunk 2]
```diff
@@ -43,6 +47,9 @@ public async Task<OperationResult> Handle(InviteUserToOrganisationCommand reques
             return ex.OperationResult;
         }
 
+        var user = createInvitationResult.ReturnedData!;
+        await _eventDispatcher.Publish(new UserInvitedEvent(user.FirstName.Value, user.LastName.Value), cancellationToken);
```
Comment (Carq) [inline src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs:51]: Publishing of event should be part of the domain entity. Check `DomainEntity` class.

**Review by Carq — APPROVED** [review]

File: src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs (hunk 2 above)
Comment (Antek012) [reply src/HE.Investments.Account.Domain/UserOrganisation/CommandHandlers/InviteUserToOrganisationCommandHandler.cs]: changed

**Pull request merged**